hardhat.config.js

```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  solidity: {
+    version: "0.8.24",
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 200,
+      }
+    },
```

Comment:
I think we should programmatically mirror the settings from foundry.toml here to keep things consistent between the two variants of consuming the repo (though I actually don't know to what extent the config here matters at all except for the verifying build step in the prepublish script).

Reply:
I made the PR quickly and could not find the current version. Of course, it would be better to align the version.

Reply:
You are right about the scope of that. It is needed only to compile the `contracts` folder to be sure that there are no issues.